pkg/disk/json.go

```diff
@@ -34,12 +34,27 @@ func WriteJWTBundleSet(jwkSet *jwtbundle.Set, dir string, jwtBundleFilename stri
 }
 
 // WriteJWTBundle write the given JWT SVID to disk
-func WriteJWTSVID(jwtSVID *jwtsvid.SVID, dir, jwtSVIDFilename string, jwtSVIDFileMode fs.FileMode) error {
+func WriteJWTSVID(jwtSVIDs []*jwtsvid.SVID, dir, jwtSVIDFilename string, jwtSVIDFileMode fs.FileMode, hint string) error {
 	filePath := path.Join(dir, jwtSVIDFilename)
-
+	jwtSVID := getJWTByHint(jwtSVIDs, hint)
+	if jwtSVID == nil {
+		return fmt.Errorf("failed to find the hinted svid")
+	}
 	return os.WriteFile(filePath, []byte(jwtSVID.Marshal()), jwtSVIDFileMode)
 }
 
+func getJWTByHint(jwtSVIDs []*jwtsvid.SVID, hint string) *jwtsvid.SVID { // Choose a better name :D
```

Comment:
The workload api has a [WithDefaultJWTSVIDPicker](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/workloadapi#WithDefaultJWTSVIDPicker) filter. What do you think of structuring this slightly differently by using this to filter svids by hint through the worklaod api. Then the logic in this library will be unchanged and simpler just to write to everything to disk.

Reply:
When I was digging through the code, I didn't find an easy way to apply that filter. Maybe I missed something though. Do you know where in the code we could do that?

Reply:
Yea youre right. Its not available through the APIs we are using.

Reply:
Marcos was talking about maybe a major refactoring at some point of the spiffe-helper. Maybe that would be a good time to consider refactoring that part of this PR?

Reply:
Would you be able to move the logic to filter by hint to pkg/sidecar.go? I think thats a better place for this code to live. If not I can do that after merging this.

Reply:
I looked into this a little bit. Would totally have to rewrite the sidecar tests / disk tests which is a good chunk of work. I'm ok doing that, but it would be nice to be able to start using this soon as well. Can we push this pr through and then rework it after release?

Reply:
Sounds good